File: module_java/src/main/java/com/boilsnow/java/DemoActivity.java
package com.boilsnow.java;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.launcher.ARouter;

/**
 * Description:
 * Remark:
 */
@Route(path = "/java/demo/a")
public class DemoActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.java_activity_main);

        findViewById(R.id.btnGO).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ARouter.getInstance().build("/kotlin/demo/a").navigation();
            }
        });
    }
}

File: module_kotlin/src/main/java/com/boilsnow/kotlin/DemoActivity.kt
package com.boilsnow.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.alibaba.android.arouter.facade.annotation.Route
import com.alibaba.android.arouter.launcher.ARouter
import kotlinx.android.synthetic.main.kotlin_activity_main.*

/**
 * Description:
 * Remark:
 */
@Route(path = "/kotlin/demo/a")
class DemoActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.kotlin_activity_main)

        btnGO.setOnClickListener {
            ARouter.getInstance().build("/java/demo/a").navigation()
        }
    }
}
